exchange/src/wrapper/algo_wrapper/binary/binary_wrapper.cpp

```diff
@@ -0,0 +1,135 @@
+#include "wrapper/algo_wrapper/binary/binary_wrapper.hpp"
+
+#include "wrapper/algo_wrapper/wrapper.hpp"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <fmt/core.h>
+#include <sys/mman.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <cerrno>
+#include <cstdio>
+#include <cstring>
+
+#include <system_error>
+
+namespace nutc {
+namespace wrapper {
+
+BinaryWrapper::BinaryWrapper(
+    std::string algo, std::string trader_id, MarketOrderFunc publish_market_order
+) : Wrapper(std::move(algo), std::move(trader_id))
+{
+    // Create a memory file descriptor
+    memfd_ = memfd_create("wrapper-library", MFD_CLOEXEC);
+    if (memfd_ == -1) {
+        throw std::system_error(errno, std::generic_category(), "memfd_create failed");
+    }
+
+    auto bytes_written = write(memfd_, algo_.data(), algo_.size());
+    if (bytes_written == -1 || static_cast<size_t>(bytes_written) != algo_.size()) {
+        close(memfd_);
+        throw std::system_error(
+            errno, std::generic_category(), "write to memfd failed"
+        );
+    }
+
+    std::string lib_path = "/proc/self/fd/" + std::to_string(memfd_);
```

Comment:
Can we not use the descriptor returned by memfd_create?

Reply:
Unfortunately, no. dlopen must take a path

Reply:
I searched around a lot for this, seems to be the only way

Reply:
Let's use helper functions to clean this up

Reply:
Re descriptor: sounds good @glasss13